resources/recursor_service_sysvinit.rb

```diff
@@ -16,51 +16,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-include ::Pdns::PdnsRecursorHelpers
-
-resource_name :pdns_recursor_service_sysvinit
 
 provides :pdns_recursor_service, os: 'linux' do |node|
   %w(debian ubuntu centos).include?(node['platform'])
 end
 
-property :instance_name, String, name_property: true
+include Pdns::RecursorHelpers
+property :instance_name, String, name_property: true, callbacks: {
+  'should not contain a hyphen' => ->(param) { !param.include?('-') },
+}
 property :cookbook, [String, NilClass], default: 'pdns'
```

Comment:
Same question as before with setting `cookbook` or `source` as nil. I think that would not cause sane behavior unless I am misunderstanding something